File: app/types/mutation-result.ts
/**
 * データ変更クエリ実行結果
 */
export interface MutationResult {
  /** 影響行数 */
  affectedRows: number
  /** 実行時間（ミリ秒） */
  executionTimeMs: number
}

/**
 * データ変更クエリ実行リクエスト
 */
export interface MutationExecuteRequest {
  /** 接続ID */
  connectionId: string
  /** 実行するSQL */
  sql: string
  /** タイムアウト（秒） */
  timeoutSeconds?: number
}

File: src-tauri/src/models/mutation_result.rs
use serde::{Deserialize, Serialize};

/// データ変更クエリ実行結果
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MutationResult {
    pub affected_rows: u64,
    pub execution_time_ms: u64,
}

/// データ変更クエリ実行リクエスト
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MutationExecuteRequest {
    pub connection_id: String,
    pub sql: String,
    pub timeout_seconds: Option<u32>,
}
